datafusion/common/src/dfschema.rs

```diff
@@ -662,12 +666,12 @@ impl DFField {
     }
 
     /// Create a qualified field from an existing Arrow field
-    pub fn from_qualified(
-        qualifier: impl Into<OwnedTableReference>,
+    pub fn from_qualified<'a>(
+        qualifier: impl Into<TableReference<'a>>,
```

Comment:
im curious about why explicit lifetime is needed here?

Reply:
Basically because the compiler told me to do so 😆 

But seriously, I think the reason is that `impl` doesn't support anonymous lifetimes -- I would have to change this to use the explicit generic syntax

Here is what happens if I remove `<'a>`:

```
   --> datafusion/common/src/dfschema.rs:113:45
    |
113 |         qualifier: impl Into<TableReference<'_>>,
    |                                             ^^ expected named lifetime parameter
    |
help: consider introducing a named lifetime parameter
    |
112 ~     pub fn try_from_qualified_schema<'a>(
113 ~         qualifier: impl Into<TableReference<'a>>,
    |
````